embark-bibtex.el

```diff
@@ -0,0 +1,150 @@
+;; - Provides instant search results as you type
+;; - Powerful search expressions
+;; - Open the PDFs, URLs, or DOIs associated with an entry
+;; - Insert LaTeX cite commands, Ebib links, or Pandoc citations,
+;;   BibTeX entries, or plain text references at point, attach PDFs to
+;;   emails
+;; - Attach notes to publications
+;;
+;; Install:
+;;
+;;   Put this file in a directory included in your load path or
+;;   install embark-bibtex from MELPA (preferred).  Then add the
+;;   following in your Emacs startup file:
+;;
+;;     (require 'embark-bibtex)
+;;
+;;   Alternatively, you can use autoload:
+;;
+;;     (autoload 'embark-bibtex "embark-bibtex" "" t)
+;;
+;;   Requirements are parsebib, embark, s, dash, and f.  The easiest way
+;;   to install these packages is through MELPA.
+;;
+;;   Let embark-bibtex know where it can find your bibliography by
+;;   setting the variable `bibtex-completion-bibliography'.  See the
+;;   manual for more details:
+;;
+;;     https://github.com/tmalsburg/helm-bibtex/blob/master/README.embark-bibtex.org
+;;
+;; Usage:
+;;
+;;    Do M-x embark-bibtex and start typing a search query when prompted.
+
+;;; Code:
+
+(require 'embark)
+(require 'marginalia)
+(require 'bibtex-completion)
+
+(defcustom embark-bibtex-default-action 'embark-bibtex-open-any
+  "The default action for the `embark-bibtex` command."
+  :group 'bibtex-completion
+  :type 'function)
+
+(defmacro embark-bibtex-embarkify-action (action name)
+  "Wraps the function ACTION in another function named NAME which
+extracts the key from the candidate selected in embark and
+passes it to ACTION."
+  `(defun ,name (cand)
+     (interactive "s ")
+     (,action (list cand))))
+
+(embark-bibtex-embarkify-action bibtex-completion-open-any embark-bibtex-open-any)
+(embark-bibtex-embarkify-action bibtex-completion-open-pdf embark-bibtex-open-pdf)
+(embark-bibtex-embarkify-action bibtex-completion-open-url-or-doi embark-bibtex-open-url-or-doi)
+(embark-bibtex-embarkify-action bibtex-completion-insert-citation embark-bibtex-insert-citation)
+(embark-bibtex-embarkify-action bibtex-completion-insert-reference embark-bibtex-insert-reference)
+(embark-bibtex-embarkify-action bibtex-completion-insert-key embark-bibtex-insert-key)
+(embark-bibtex-embarkify-action bibtex-completion-insert-bibtex embark-bibtex-insert-bibtex)
+(embark-bibtex-embarkify-action bibtex-completion-add-PDF-attachment embark-bibtex-add-PDF-attachment)
+(embark-bibtex-embarkify-action bibtex-completion-edit-notes embark-bibtex-edit-notes)
+(embark-bibtex-embarkify-action bibtex-completion-show-entry embark-bibtex-show-entry)
+(embark-bibtex-embarkify-action bibtex-completion-add-pdf-to-library embark-bibtex-add-pdf-to-library)
+
+(defun embark-bibtex--get-candidates ()
+  "Return all keys from bibtex-completion-candidates."
+  (mapcar (lambda (cand)
+            (cdr (assoc "=key=" cand)))
+          (bibtex-completion-candidates)))
+
+;;;###autoload
+(defun embark-bibtex (bib-entry)
```

Comment:
```suggestion
(defun embark-bibtex (local-bib)
```
Minor thing here, but `bib-entry` suggests a single item, rather than the full collection. This suggests uses the same name as `ivy-bibtex` and `helm-bibtex`.